src/OrchardCore/OrchardCore.Roles.Core/CommonPermissions.cs

```diff
@@ -6,13 +6,16 @@ public static class CommonPermissions
 {
     public static readonly Permission ManageRoles = new("ManageRoles", "Managing Roles", isSecurityCritical: true);
 
+    [Obsolete("This Permission is no longer used and will be removed. Instead use OrchardCore.Users.CommonPermissions.AssignRoleToUsers.")]
     public static readonly Permission AssignRoles = new("AssignRoles", "Assign Roles", [ManageRoles], isSecurityCritical: true);
 
     /// <summary>
     /// Dynamic permission template for assign role.
     /// </summary>
+    [Obsolete("This Permission is no longer used and will be removed. Instead use OrchardCore.Users.CommonPermissions.CreateAssignRoleToUsersPermission(roleName).")]
```

Comment:
This is on a private field, so not needed.

Reply:
It is needed to avoid the warning otherwise the build fails

Reply:
That warning is better fixed with a pragma, since the attribute is really not doing anything here, but whatever.